paasta_tools/vitesscluster_tools.py

```diff
@@ -72,21 +688,62 @@ def validate(
             return []
 
 
+def generate_vitess_instance_config(
+    instance_config: VitessDeploymentConfigDict,
+) -> VitessDeploymentConfigDict:
```

Comment:
on another pass of this: i'm a little confused by what we're doing here: we're taking a `VitessDeploymentConfigDict` and returning a `VitessDeploymentConfigDict`? is this necessary? can we not fully construct the original `VitessDeploymentConfigDict` when that gets created?

Reply:
https://github.com/Yelp/paasta/pull/3889/commits/48ebe609e02377336248dcbb9bfed29a0d6213ae